app/packages/core/src/components/Actions/ActionsRow.tsx

```diff
@@ -162,7 +162,9 @@ const Tag = ({
   const [available, setAvailable] = useState(true);
   const labels = useRecoilValue(fos.selectedLabelIds);
   const samples = useRecoilValue(fos.selectedSamples);
+  const canTag = useRecoilValue(fos.canTagSamplesOrLabels);
   const readOnly = useRecoilValue(fos.readOnly);
+  const isReadOnly = readOnly || !canTag;
```

Comment:
minor - feels confusing having `isReadOnly` deriving from `readOnly` and another condition.

Can we instead bake in the `!canTag` as part of `readOnly`? so this line will become
```
const isReadOnly = readOnly;
```
or can be completely and readOnly instead being used.

Reply:
Ibrahim suggested that we keep the `readOnly` name from reading in the state. Meanwhile, `!canTag` is also necessary in this `isReadOnly`, because `readOnly` right now reflects `isSnapshot` (in the future it will possibly capture more).

Reply:
Good point. I believe in some parts of the app where the combination of readOnly and permission atom is used, we use `canEdit` so that's a possible alternative. For example:

```js
const canEdit = !readOnly && canTag;
// Then, we would invert all the usage below
```